src/isl/util.rs

```diff
@@ -546,3 +687,30 @@ impl Annotation {
         }
     }
 }
+
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+pub enum TimestampPrecisionValue {
+    Year = -4,
+    Month = -3,
+    Day = -2,
+    Minute = -1,
+    Second = 0,
+    Millisecond = 3,
+    Microsecond = 6,
+    Nanosecond = 9,
+}
```

Comment:
This works, but I think it would be safer and a bit more self-describing to do something like this:

```rust
pub enum TimestampPrecisionValue {
    Year,
    Month,
    Day,
    Minute,
    Second,
    Millisecond,
    Microsecond,
    Nanosecond,
}

impl TimestampPrecisionValue {
  pub const fn exponent(&self) -> i64 {
    match self {
      Year => -4,
      Month = -3,
      // ... and so on.
    }
  }
}
```
(Maybe there's a better name than `exponent`?)

It's obvious to me what the subsecond precision values do, but I don't know what `Minute=-1`, `Day=-2`, etc means. If those aren't being used meaningfully anywhere, then `exponent` could return `Option<i64>` (or an unsigned type if we don't need negatives.)

Reply:
I will rename this to be `scale` as that would make more sense than `from_timestamp`.

Reply:
In the updated version you've added a `scale` method, but we're still using `Year = -4, Month = -3...` in the enum definition. I'd like to avoid that and instead define the `i64` values inside the body of the `scale` method. If we use the current system, it makes `Year = -4` part of the public API even though that value is arbitrary.